# Assignment 1

### Introduction, Data Types, Data Structures, Arithmetic Operators, Conditional Operators, Assignment Operator, Relational Operator, Arithmetic Operator Programs

### 1. Create three objects/variables x, y, z by assigning the values of 20, 30, 100

In [1]:
x,y,z = 20,30,100

In [2]:
x

20

In [3]:
y

30

In [4]:
z

100

### 2. Apply all arithmetic operations (+, -, *, /, //, %, **) using above values.


In [5]:
x+y

50

In [6]:
x-z

-80

In [7]:
y*z

3000

In [8]:
x/z

0.2

In [9]:
z//y

3

In [10]:
z%y

10

In [11]:
x**y

1073741824000000000000000000000000000000

### 3. Write a program to print whether the given number is odd or even, use if-else conditional statement

In [12]:
a = int(input("Enter the number :"))

if a % 2 == 0:
    print('even')
else:
    print('odd')

Enter the number :33
odd


###  1) Solve the following with the steps in Python

*  Use PAMDAS method for this

In [13]:
6 + 4 * 9 % (3 + 1) / 6 - 2

4.0

In [14]:
11 + 3**(1 + 3) % 3 * 3 - 3

8

In [15]:
7 + 4 * 6 % (5 + 1) / 6 + 3

10.0

In [16]:
6 + 6 * (4+6) // 7 / 4 - 1

7.0

In [17]:
12 + 6* (7+3) // 7 / 2 - 3

13.0

In [18]:
11 + 4 * 2 ** (3 + 1) % 4 - 3

8

###  2) Give the AND operations for the following in “TRUE” and “FALSE” Context and validate in Python:

* The AND operator should show “TRUE” for the first set of Validations And “FALSE” for the second set of validations

In [19]:
x = 256 
y = 251

In [20]:
print(bool(x & y))

False


In [21]:
a = 123
b = 859

In [22]:
print(bool(a & b))

True


In [23]:
p = 89
q = 85

In [24]:
print(bool(q & b))

True


In [25]:
a = 125
b = 256
c = 324
d = 648

In [26]:
print(bool(a & b))

False


In [27]:
print(bool(b & c))

True


In [28]:
print(bool(c & d))

False


In [29]:
print(bool(d & a))

True


In [30]:
print(bool(a & c))

True


### 3) Write Python Programs for the following Arithmetic Operators:

#### a) Assign numbers 12 and 14 to variables ‘a’ and ‘b’ and add them using output variable ‘c’

In [31]:
a = 12
b = 14

In [32]:
c = a + b
c

26

#### b) Assign numbers 125, 15 and 258 to variables ‘a’, ‘b’ and ‘z’ and add them using output variable ‘d’

In [33]:
a = 125
b = 15 
c = 258

In [34]:
d = a + b + c
d

398

#### c) Assign numbers 245 and 68 to variables ‘p’ and ‘q’ and subtract them using output variable ‘t’

In [35]:
p = 245
q = 68

In [36]:
t = p - q
t

177

# Assignment 2

### Using for loop write a program to generate a table of number 14 for range 16

In [37]:
number = 14
range_limit = 16

for i in range(1, range_limit + 1):
    print(f"{number} x {i} = {number * i}")

14 x 1 = 14
14 x 2 = 28
14 x 3 = 42
14 x 4 = 56
14 x 5 = 70
14 x 6 = 84
14 x 7 = 98
14 x 8 = 112
14 x 9 = 126
14 x 10 = 140
14 x 11 = 154
14 x 12 = 168
14 x 13 = 182
14 x 14 = 196
14 x 15 = 210
14 x 16 = 224


### Using for loop write a program to generate a table of number 18 for range starting from 12 to 23

In [38]:
number = 18
start_range = 23

for i in range(12, start_range + 1):
    print(f"{number} x {i} = {number * i}")

18 x 12 = 216
18 x 13 = 234
18 x 14 = 252
18 x 15 = 270
18 x 16 = 288
18 x 17 = 306
18 x 18 = 324
18 x 19 = 342
18 x 20 = 360
18 x 21 = 378
18 x 22 = 396
18 x 23 = 414


### Create multiplication table which can take input from user for the number 16 and range 8

In [90]:
number = int(input("Enter the number :"))
range_limit = 8

for i in range(1, range_limit + 1):
    print(f"{number} x {i} = {number * i}")

Enter the number :16
16 x 1 = 16
16 x 2 = 32
16 x 3 = 48
16 x 4 = 64
16 x 5 = 80
16 x 6 = 96
16 x 7 = 112
16 x 8 = 128


### Write a program to display if a number is odd or even using if and else statement

In [40]:
n = int(input("Enter the number :"))

if n % 2 != 0:
    print("Odd")
else:
    print("Even")

Enter the number :2
Even


### Write a program to display if a number is lesser than or greater than using if and else statement

In [41]:
number = float(input("Enter the number :"))

specified_value = 20

if number < specified_value:
    print(f"{number} is less than {specified_value}.")
else:
    print(f"{number} is greater than {specified_value}.")

Enter the number :22
22.0 is greater than 20.


### Write a program of display if a number is lesser than or greater than or equal to using if-elif-else statement

In [42]:
number = float(input("Enter a number: "))

specified_value = 10

if number < specified_value:
    print(f"{number} is less than {specified_value}.")
elif number > specified_value:
    print(f"{number} is greater than {specified_value}.")
else:
    print(f"{number} is equal to {specified_value}.")


Enter a number: 25
25.0 is greater than 10.


### Create a list of numbers from 150 to 275 with a step size of 10

In [43]:
numbers = [i for i in range(150, 276, 10)]
print(numbers)

[150, 160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260, 270]


In [44]:
number = list(range(150,270,10))
number

[150, 160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260]

### Create a tuple of numbers from 345 to 485 with a step size of 15

In [45]:
num = tuple(range(345,485,15))
num

(345, 360, 375, 390, 405, 420, 435, 450, 465, 480)

### Create multiplication table which can take input from user for table number 14 and table length 12.

In [89]:
number = int(input("Enter the number :"))
range_limit = 12

for i in range(1, range_limit + 1):
    print(f"{number} x {i} = {number * i}")

Enter the number :14
14 x 1 = 14
14 x 2 = 28
14 x 3 = 42
14 x 4 = 56
14 x 5 = 70
14 x 6 = 84
14 x 7 = 98
14 x 8 = 112
14 x 9 = 126
14 x 10 = 140
14 x 11 = 154
14 x 12 = 168


# Assignment 3

* For Loops, While Loops, If-Else Statements, Tuple, Lists, Dictionaries

### Print the list of values between 120 & 250 which are divisible by 5

In [47]:
divisible_by_5 = [num for num in range(120, 251) if num % 5 == 0]

print(divisible_by_5)


[120, 125, 130, 135, 140, 145, 150, 155, 160, 165, 170, 175, 180, 185, 190, 195, 200, 205, 210, 215, 220, 225, 230, 235, 240, 245, 250]


### Create a tuple with the values ('che', 'blr', 'ggn', 'noi', 'che', 'ggn', 'noi', 'che', 'ggn', 'noi', 'che')

In [48]:
t_1 = ('che', 'blr', 'ggn', 'noi', 'che', 'ggn', 'noi', 'che', 'ggn', 'noi', 'che')
t_1

('che', 'blr', 'ggn', 'noi', 'che', 'ggn', 'noi', 'che', 'ggn', 'noi', 'che')

In [49]:
type(t_1)

tuple

### Find the number of times ggn is repeated in the above tuple using methods

In [50]:
t_1.count("ggn")

3

### Create a tuple with values between 100, and 200 and extract values from 150 to 200 from the list using slicing & dicing

In [51]:
tup = list(range(100,201))
print(tup, end=",")

[100, 101, 102, 103, 104, 105, 106, 107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119, 120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132, 133, 134, 135, 136, 137, 138, 139, 140, 141, 142, 143, 144, 145, 146, 147, 148, 149, 150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200],

In [52]:
print(tup[50:101], end=",")

[150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200],

In [53]:
# Creating a tuple with values between 100 and 200
my_tuple = tuple(range(100, 201))

# Extracting values from 150 to 200 using slicing
extracted_values = my_tuple[50:101]  # Index 50 corresponds to 150, and index 100 corresponds to 200

# Printing the extracted values
print(extracted_values)


(150, 151, 152, 153, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163, 164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180, 181, 182, 183, 184, 185, 186, 187, 188, 189, 190, 191, 192, 193, 194, 195, 196, 197, 198, 199, 200)


### Print the list of values between 780 & 920 which are divisible by 6

In [54]:
divisible_by_6 = [num for num in range(780, 920) if num % 6 == 0]

print(divisible_by_6)

[780, 786, 792, 798, 804, 810, 816, 822, 828, 834, 840, 846, 852, 858, 864, 870, 876, 882, 888, 894, 900, 906, 912, 918]


### Write a Python program which accepts the radius of a circle from the user and compute the area.
Sample Output:
r = 1.1
Area = 3.8013271108436504

In [55]:
import math
radius = float(input("Enter the radius :"))

area = math.pi * radius ** 2

print(radius)
print(area)

Enter the radius :1.1
1.1
3.8013271108436504


### Write a Python program which accepts the user's first and last name and print them in reverse order with a space between them.

In [56]:
def my_function(x):
    return x[::-1]    # we use slicing for reverse the str

mytxt = my_function("Yatharth Pandya")

print(mytxt)

aydnaP htrahtaY


### Write a Python program which accepts a sequence of comma-separated numbers from user and generate a list and a tuple with those numbers.
Sample data : 3, 5, 7, 23


Output:
List : ['3', ' 5', ' 7', ' 23']
Tuple : ('3', ' 5', ' 7', ' 23')

In [57]:
input_s = input("Enter comma-separated numbers :")

num_list = input_s.split(",")

num_tuple = tuple(num_list)

print("List :", num_list)
print("Tuple :", num_tuple)

Enter comma-separated numbers :3,5,7,23
List : ['3', '5', '7', '23']
Tuple : ('3', '5', '7', '23')


### Write a Python program to display the first and last colors from the following list.
color_list = ["Red","Green","White" ,"Black"]

In [58]:
color_list = ["Red","Green","White" ,"Black"]
color_list[0:4:3]

['Red', 'Black']

### Write a Python program to calculate number of days between two dates.

Sample date : (2014, 7, 2), (2014, 7, 11), Expected output : 9 days

In [59]:
from datetime import datetime

date1 = datetime(2014, 7, 2)
date2 = datetime(2014, 7, 11)

# Difference
difference = date2 - date1

# The difference between 2 dates
number_of_days = difference.days

print("Number of days between the two dates:", number_of_days, "days")

Number of days between the two dates: 9 days


### Write a Python program to get the area of a triangle with by taking inputs from user.

In [60]:
height = float(input("Enter the height of triangle :"))
base = float(input("Enter the base of tringle :"))

tri_area = 0.5 * base * height
print(tri_area)

Enter the height of triangle :2
Enter the base of tringle :23
23.0


### Write a Python program to obtain the area of a square, circle and rectangle by taking inputs from user

In [61]:
a = float(input("Enter side  :"))
r = float(input("Enter radius :"))
l = float(input("Enter length :"))
w = float(input("Enter weight :"))

# Squre
square_area = a ** 2

# Rectangle
rectangle_area = l * w

# Circle
circle_area = math.pi * r ** 2

print(square_area)
print(rectangle_area)
print(circle_area)

Enter side  :2
Enter radius :3
Enter length :4
Enter weight :5
4.0
20.0
28.274333882308138


### Print the list of values between 500 & 700 which are divisible by 7

In [62]:
div_by_7 = [num for num in range(500, 700) if num % 7 == 0]
print(div_by_7)

[504, 511, 518, 525, 532, 539, 546, 553, 560, 567, 574, 581, 588, 595, 602, 609, 616, 623, 630, 637, 644, 651, 658, 665, 672, 679, 686, 693]


### Write a Python program to get the volume of a sphere with radius 6.

In [63]:
r = 6

sphere_volume = 4/3*3.14*r**3
print(sphere_volume)

904.3199999999999


# Assignment 4

* Control Statements, Conditional Statements, Programs on For Loops, While Loops, Programs on if-else, if-elif-else statements, Tuple, Lists, Dictionaries on Arguments, Keyword Arguments, User Defined Functions

### Write a program which will find all numbers between 1500 and 2600 which are divisible by 7 but are not a multiple of 5.

In [64]:
numbers = []

for i in range(1500,2600):
    if i % 7 == 0 and i % 5 != 0:
        numbers.append(i)

print(numbers)

[1512, 1519, 1526, 1533, 1547, 1554, 1561, 1568, 1582, 1589, 1596, 1603, 1617, 1624, 1631, 1638, 1652, 1659, 1666, 1673, 1687, 1694, 1701, 1708, 1722, 1729, 1736, 1743, 1757, 1764, 1771, 1778, 1792, 1799, 1806, 1813, 1827, 1834, 1841, 1848, 1862, 1869, 1876, 1883, 1897, 1904, 1911, 1918, 1932, 1939, 1946, 1953, 1967, 1974, 1981, 1988, 2002, 2009, 2016, 2023, 2037, 2044, 2051, 2058, 2072, 2079, 2086, 2093, 2107, 2114, 2121, 2128, 2142, 2149, 2156, 2163, 2177, 2184, 2191, 2198, 2212, 2219, 2226, 2233, 2247, 2254, 2261, 2268, 2282, 2289, 2296, 2303, 2317, 2324, 2331, 2338, 2352, 2359, 2366, 2373, 2387, 2394, 2401, 2408, 2422, 2429, 2436, 2443, 2457, 2464, 2471, 2478, 2492, 2499, 2506, 2513, 2527, 2534, 2541, 2548, 2562, 2569, 2576, 2583, 2597]


### Write a program which will count all string values in the list [4, 6, 'Python', '10', 1, 'SQL', 5.5, True, 3, 'John', None, 7, 'How']

In [65]:
l_1 = [4, 6, 'Python', '10', 1, 'SQL', 5.5, True, 3, 'John', None, 7, 'How']
l_1

[4, 6, 'Python', '10', 1, 'SQL', 5.5, True, 3, 'John', None, 7, 'How']

In [66]:
string_count = 0

for element in l_1:
    # Check if the element is an instance of str (string)
    if isinstance(element, str):
        string_count += 1
        
print(string_count)

5


### Write a program which will find all numbers between 2500 and 3600 which are divisible by 6 but are not a multiple of 3.

In [69]:
numbers = []

for i in range(2500, 3601):
    if i % 6 == 0 and i % 3 != 0:
        numbers.append(i)

print(numbers)

[]


### Write a program which will count all integer values in the list [“JUST”, 6, 'Py', '10', 1, 'SQL', 5.5, False, 6, 'John', None, 7, 'Which']

In [71]:
l_2 = ['JUST', 6, 'Py', '10', 1, 'SQL', 5.5, False, 6, 'John', None, 7, 'Which']
l_2

['JUST', 6, 'Py', '10', 1, 'SQL', 5.5, False, 6, 'John', None, 7, 'Which']

In [75]:
int_value = 0

for element in l_2:
    if isinstance(element, int):
        int_value += 1
        
print(int_value)

5


### Write a program which will count all float values in the list [6, 'The', '10.2', 1.1, 'SQL', 5.5, False, 6.8, 'John', None, 7, 'Mars']

In [73]:
l_3 =  [6, 'The', '10.2', 1.1, 'SQL', 5.5, False, 6.8, 'John', None, 7, 'Mars']
l_3

[6, 'The', '10.2', 1.1, 'SQL', 5.5, False, 6.8, 'John', None, 7, 'Mars']

In [77]:
float_value = 0

for element in l_3:
    if isinstance(element, float):
        float_value += 1
        
print(float_value)

3


### Write a Python program which takes the input from a list of numbers, squares them and adds them to another list. Perform the same using list comprehension also.

In [78]:
# use Loop

numbers = [2,3,4,5,6]

squared_num = []

for num in numbers:
    squared_num.append(num ** 2)
    
print(squared_num)

[4, 9, 16, 25, 36]


In [82]:
# Use list comprehension

numbers = [2,3,4,5,6]

squared_num = [num ** 2 for num in numbers]

print(squared_num)

[4, 9, 16, 25, 36]


### Write a Python program which takes the input from a list of numbers, cubes them and adds them to another list. Perform the same using list comprehension also

In [83]:
# Use list comprehension

numbers = [11,13,15,16,17]

cubes_num = [num ** 3 for num in numbers]

print(cubes_num)

[1331, 2197, 3375, 4096, 4913]


### Write a Python program which takes the input from a list (use range function) of numbers, cubes them and adds them to another list. Perform the same using list comprehension also.

In [85]:
numbers = list(range(1, 11))

cubed_numbers = []

for num in numbers:
    cubed_numbers.append(num ** 3)
    
print(cubed_numbers)

[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]


In [87]:
# List Comprehension

numbers = list(range(1,11))

cubed_num = [num ** 3 for num in numbers]

print(cubed_num)

[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]


### Write a Python program which takes the input from a list (use range function) of numbers, squares them and adds them to another list. Perform the same using list comprehension also.

In [88]:
# List Comprehension

numbers = list(range(11,21))

squared_num = [num ** 2 for num in numbers]

print(squared_num)

[121, 144, 169, 196, 225, 256, 289, 324, 361, 400]


### Write a program that takes a string and returns reversed string. i.e. if input is "pqrs4567" output should be "7654srqp"

In [84]:
str_1 = ('pqrs4567')

print(str_1[::-1])

7654srqp
